scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -3877,6 +3950,48 @@ def __gather_metrics_from_api(self, containers, cluster_name):
                     k8s_extra = {}
             self.__gather_metrics_from_api_for_container(info["name"], k8s_extra)
 
+    def __gather_metrics_from_kubelet(self, containers, kubelet_api, cluster_name):
+        cluster_info = self.__get_cluster_info(cluster_name)
+
+        try:
+            stats = kubelet_api.query_stats()
+
+            all_k8s_extra = {}
+            containers_to_check = {}
+            for cid, info in six.iteritems(containers):
+                containers_to_check[info["name"]] = {}
+                if self.__include_controller_info:
+                    k8s_extra = self.__get_k8s_controller_info(info)
+                    if k8s_extra is not None:
+                        k8s_extra.update(cluster_info)
+                        k8s_extra.update({"pod_uid": info["name"]})
+                        all_k8s_extra[info["name"]] = k8s_extra
+
+            pods = stats.get("pods", [])
+
+            for pod in pods:
+                container_stats = pod.get("containers", {})
```

Comment:
Is this attribute value a dictionary or list of dictionaries?

Based on the code, it looks like it's probably a list of dicts, but then default value we pass to ``pod.get`` is likely wrong.

Reply:
It is a dictionary, the same process of iterating over `pods` and then doing `pod.get()` was already done in the existing `__gather_k8s_metrics_from_kubelet` method, and it works properly from my testing.

Reply:
It is a dictionary, the same process of iterating over pods and then doing `pod.get()` was already done in the existing `__gather_k8s_metrics_from_kubelet` method, and it works properly from my testing.